dds/DCPS/XTypes/TypeObject.cpp

```diff
@@ -4858,6 +4858,12 @@ bool operator>>(Serializer& strm, XTypes::TypeIdentifierPair& stru)
     && (strm >> stru.type_identifier2);
 }
 
+bool to_type_object(const unsigned char* buffer, size_t size, XTypes::TypeObject& to)
+{
+  OpenDDS::DCPS::Message_Block_Shared_Ptr message_block(new ACE_Message_Block(reinterpret_cast<const char*>(buffer), size));
```

Comment:
This can use an ACE_Message_Block on the stack.  There should be examples of that in the repo, let me know if I can help track one down